freemocap/core_processes/post_process_skeleton_data/calculate_center_of_mass.py

```diff
@@ -0,0 +1,383 @@
+                    trunk_virtual_markers[1],
+                ]
+            elif segment == "left_hand" or segment == "right_hand":
+                proximal_joint_hand = segment_info["Joint_Connection"][0]
+                if segment == "left_hand":
+                    distal_joint_hand = "left_index"
+                else:
+                    distal_joint_hand = "right_index"
+
+                proximal_joint_hand_index = mediapipe_indices.index(proximal_joint_hand)
+                distal_joint_hand_index = mediapipe_indices.index(distal_joint_hand)
+
+                mediapipe_pose_skeleton_coordinates[segment] = [
+                    mediapipe_pose_data[frame, proximal_joint_hand_index, :],
+                    mediapipe_pose_data[frame, distal_joint_hand_index, :],
+                ]
+
+            elif segment == "left_foot" or segment == "right_foot":
+                if segment == "left_foot":
+                    proximal_joint_foot_name = "left_ankle"
+                else:
+                    proximal_joint_foot_name = "right_ankle"
+
+                proximal_joint_foot_index = mediapipe_indices.index(proximal_joint_foot_name)
+
+                distal_joint_foot = segment_info["Joint_Connection"][1]
+                distal_joint_foot_index = mediapipe_indices.index(distal_joint_foot)
+                mediapipe_pose_skeleton_coordinates[segment] = [
+                    mediapipe_pose_data[frame, proximal_joint_foot_index, :],
+                    mediapipe_pose_data[frame, distal_joint_foot_index, :],
+                ]
+
+            else:
+                proximal_joint_name = segment_info["Joint_Connection"][0]
+                distal_joint_name = segment_info["Joint_Connection"][1]
+
+                # get the mediapipe index for the proximal and distal joint for this segment
+                proximal_joint_index = mediapipe_indices.index(proximal_joint_name)
+                distal_joint_index = mediapipe_indices.index(distal_joint_name)
+
+                # use the mediapipe indices to get the XYZ coordinates for the prox/distal joints and throw it in a dictionary
+                # mediapipe_pose_skeleton_coordinates[segment] = {'proximal':mediapipe_pose_data[frame,proximal_joint_index,:],'distal':mediapipe_pose_data[frame,distal_joint_index,:]}
+                mediapipe_pose_skeleton_coordinates[segment] = [
+                    mediapipe_pose_data[frame, proximal_joint_index, :],
+                    mediapipe_pose_data[frame, distal_joint_index, :],
+                ]
+
+        mediapipe_frame_segment_joint_XYZ.append(mediapipe_pose_skeleton_coordinates)
+        f = 2
+
+    return mediapipe_frame_segment_joint_XYZ
+
+
+# %%
+# values for segment weight and segment mass percentages taken from Winter anthropometry tables
+# https://imgur.com/a/aD74j
+# Winter, D.A. (2005) Biomechanics and Motor Control of Human Movement. 3rd Edition, John Wiley & Sons, Inc., Hoboken.
+
+
+BODY_SEGMENT_NAMES  = [
```

Comment:
Disclaimer: I haven't looked at this part toooo deeply, but I had a sloppy thought, re: "calculating COM from arbitrary skeleton input" 

You could probably build a dictionary (or Pydantic model) that uses this list as the `keys` and something akin to the model in `freemocap.data_layer.data_saver.data_models.VirtualMarkerDefinition`  

(For reference, that's a Pydantic version of the `mediapipe_virtual_marker_definitions_dict` defined here as a good ol' fashioned dictionary:` freemocap/data_layer/export_data/blender_stuff/blender_bpy_export_scripts/freemocap_blender_megascript_take2.py:788`)

That way, you could have a concise schema (save and loadable as a `json`) that could define how to combine the `mediapipe` markers to get the joint_centers necessary to calcuate the COM - i.e. it would be a compact way to define the interface between 'mediapipe shaped data' and the dataframe defined in the `build_anthropometric_dataframe()` function (which can also be saved/loaded as a json or csv)

Think about whether that makes sense and we can chat about it later :)

Reply:
Oh, just realized - the above story might want you to make another list called `BODY_JOINT_CENTERS` or something

Reply:
...which could similarly be a dictionary with values corresponding to your `joint_connections ` variable below